### PR TITLE
Implemeted ResponseCache using Client and Proxy

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9082](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9082)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Implemeted ResponseCache using Client and Proxy

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/WebClient/src/Server/Controllers/UserProfileController.cs [hunk 1]
```diff
@@ -184,6 +184,7 @@ public IActionResult RecoverUserProfile(string hdid)
         [HttpGet]
         [Route("termsofservice")]
         [AllowAnonymous]
+        [ResponseCache(Location = ResponseCacheLocation.Any, Duration = 3600)]
```
Comment (leobcgov) [inline Apps/WebClient/src/Server/Controllers/UserProfileController.cs:187]: - On the top of caching at the client and proxy level, I think the best method of caching is to use the in-memory cache. However, as we don't expect that the TermsOfService page will be requested very often, so it doesn't make sense to loose memory to cache it in this scenario.
- ref. https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response

**Review by FuriousLlama — APPROVED** [review]

File: Apps/WebClient/src/Server/Controllers/UserProfileController.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/Server/Controllers/UserProfileController.cs:187]: pretty cool. Have you checked if the response header says is cached?

Comment (leobcgov) [reply Apps/WebClient/src/Server/Controllers/UserProfileController.cs:187]: I run the WebClient in VS 2019 debug mode then open the ToS page in Chrome then click F5. The first load should call the controller method. The 2nd one should not.

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/WebClient/src/Server/Controllers/UserProfileController.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/Server/Controllers/UserProfileController.cs:187]: yeah, the header response on chrome dev tools would also say is cache, but that is a good test too!

Comment (leobcgov) [reply Apps/WebClient/src/Server/Controllers/UserProfileController.cs:187]: ah great to know this header response, thanks @FuriousLlama

**Pull request merged**